views/utilities.js

```diff
@@ -227,6 +227,39 @@ module.exports  = {
         return requestData;
     },
 
+    // Searches for any previous request to see if this is a duplicate
+    // Returns null if not found, otherwise returns first record
+    findPriorRequest: function(requestData) {
+      // Replacing non-numeric characters with Postgres match operator, adding 
+      // one to the beginning to accommodate country codes
+      var phoneQuery = "%" + requestData.phone.replace(/[^0-9]/g, "%");
+
+      return db.Request.findOne({
+          where: {
+            name: { $ilike: requestData.name },
+            source: requestData.source,
+            address: { $ilike: requestData.street_address },
+            city: { $ilike: requestData.city },
+            state: { $ilike: requestData.state },
+            zip: requestData.zip_final,
+            email: { $ilike: requestData.email },
+            phone: { $like: phoneQuery },
+          }
+      });
+    },
+
+    // Force update of updatedAt without changing attributes
+    // source: https://github.com/sequelize/sequelize/issues/3759
+    updateRequestTime: function(request) {
+      request.changed('updatedAt', true);
+      return request.save();
+    },
```

Comment:
This is cool!

Reply:
I guess the only concern would be if it might not continue to work through later versions of Sequelize?  Any thoughts about that?

Reply:
Hmm, from the PR comments it looks like they may add a better way of doing this, but that this is using internal methods that don't look like they'll be deprecated. The only other way I've seen is to update a property that doesn't actually exist on the instance